From: Jan Kara <jack@suse.cz>
Subject: ext3: Increase maximum reservation window
References: bnc#716023
Patch-mainline: Never

With 3.0 kernel, CFQ and writeback end up being more fair to tiobench threads
and thus when multiple threads are running their files end up more interleaved.
In particular, the average continuous extent size now ends up being somewhere
around half of the maximum allocation reservation window where previously it
has been even several reservation windows. Thus read performance suffers
noticeably (more than 15%).

We mitigate the issue by increasing the maximum size of allocation reservation
window which restores the performance. It should not influence non-sequential
kind of workloads since allocation window is sized dynamically depending on the
hit ratio of the current allocation window.

Signed-off-by: Jan Kara <jack@suse.cz>

diff -rupX /crypted/home/jack/.kerndiffexclude linux-3.0-SLE11-SP2/include/linux/ext3_fs.h linux-3.0-SLE11-SP2-ext3rsv/include/linux/ext3_fs.h
--- a/include/linux/ext3_fs.h	2011-10-17 21:57:07.588149289 +0200
+++ b/include/linux/ext3_fs.h	2011-10-21 14:49:14.266036164 +0200
@@ -33,7 +33,7 @@
  */
 #define EXT3_DEFAULT_RESERVE_BLOCKS     8
 /*max window size: 1024(direct blocks) + 3([t,d]indirect blocks) */
-#define EXT3_MAX_RESERVE_BLOCKS         1027
+#define EXT3_MAX_RESERVE_BLOCKS         (4*1027)
 #define EXT3_RESERVE_WINDOW_NOT_ALLOCATED 0
 
 /*
